lua/telescope/actions.lua

```diff
@@ -165,6 +168,14 @@ actions.set_command_line = function(prompt_bufnr)
   vim.cmd(entry.value)
 end
 
+actions.run_builtin = function(prompt_bufnr)
+  local entry = actions.get_selected_entry(prompt_bufnr)
+
+  actions.close(prompt_bufnr, false)
```

Comment:
instead of doing this (which doesn't follow the pattern of actions) you can just do `vim.cmd [[startinsert]]` right here.